public/app/features/variables/state/transactionReducer.ts

```diff
@@ -32,10 +46,29 @@ const transactionSlice = createSlice({
     variablesClearTransaction: (state, action: PayloadAction<undefined>) => {
       state.uid = null;
       state.status = TransactionStatus.NotStarted;
+      state.isDirty = false;
     },
   },
+  extraReducers: (builder) =>
+    builder.addMatcher(actionAffectsDirtyState, (state, action) => {
+      if (state.status === TransactionStatus.Completed) {
+        state.isDirty = true;
+      }
+    }),
 });
 
+function actionAffectsDirtyState(action: AnyAction): boolean {
+  return [
+    removeVariable.type,
+    addVariable.type,
+    changeVariableProp.type,
+    changeVariableOrder.type,
+    duplicateVariable.type,
+    changeVariableType.type,
+    setCurrentVariableValue.type,
```

Comment:
Not sure changing value should make it dirty, if I remember correctly the current diff change tracker ignores current value ?

Reply:
Good point I'll remove it, if the need arises it's easy to add back.